src/frontends/pytorch/src/op/index_fill_.cpp

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+
+#include "openvino/core/node.hpp"
+#include "openvino/core/node_output.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/op/broadcast.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/convert_like.hpp"
+#include "openvino/op/reshape.hpp"
+#include "openvino/op/scatter_elements_update.hpp"
+#include "openvino/op/shape_of.hpp"
+#include "openvino/op/slice.hpp"
+#include "utils.hpp"
+
+using namespace ov::op;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_index_fill_(const NodeContext& context) {
+    // aten::index_fill_(self, dim, index, value) --> Tensor
+    num_inputs_check(context, 4, 4);
+    auto input = context.get_input(0);
+    auto dim = context.get_input(1);
+    auto index = context.get_input(2);
+    auto value = context.get_input(3);
+
+    auto const_1_vec = v0::Constant::create(element::i32, Shape{1}, {1});
+
+    auto tensor_rank = std::get<1>(get_shape_rank(context, input, true));
+    auto tensor_rank_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(tensor_rank, dim));
+    auto positive_dim = normalize_axis(context, dim, tensor_rank_correct_type);
+
+    // scalar to vec
+    auto dim_vec = context.mark_node(std::make_shared<v1::Reshape>(positive_dim, const_1_vec, false));
```

Comment:
```suggestion
    auto tensor_rank = std::get<1>(get_shape_rank(context, input, false));
    auto tensor_rank_correct_type = context.mark_node(std::make_shared<v1::ConvertLike>(tensor_rank, dim));
    auto dim_vec = normalize_axis(context, dim, tensor_rank_correct_type);

    // scalar to vec
```
You can skip `Reshape` to 1D if you get rank as 1D